astropy/io/fits/column.py

```diff
@@ -1794,10 +1794,10 @@ def __sub__(self, other):
         if not isinstance(other, (list, tuple)):
             other = [other]
         _other = [_get_index(self.names, key) for key in other]
-        indx = list(range(len(self)))
+        index = list(range(len(self)))
```

Comment:
Is this really necessary?

Reply:
It doesn't overwrite other variables, so 🤷.